code/modules/reagents/chemistry/reagents/medicine.dm

```diff
@@ -1035,3 +1035,42 @@
 	color = "#F5F5F5"
 
 // This reagent's effects are handled in heart attack handling code
+
+/datum/reagent/medicine/nanocalcium
+	name = "Nano-Calcium"
+	id = "nanocalcium"
+	description = "Highly advanced nanites equipped with calcium payloads designed to repair bones without surgery suspended in synthetic blood plasma."
+	color = "#9b3401"
+	metabolization_rate = 0.5
+
+/datum/reagent/medicine/nanocalcium/on_mob_life(mob/living/carbon/human/M)
+	switch(current_cycle)
+		if(1 to 19)
+			M.AdjustJitter(4)
+			if(prob(10))
+				to_chat(M, "<span class='notice'>Your skin feels hot and your veins are on fire!</span>")
+		if(20 to 43)
+			//If they have stimulants or stimulant drugs then just apply toxin damage instead.
+			if(M.reagents.has_reagent("methamphetamine") || M.reagents.has_reagent("crank") || M.reagents.has_reagent("bath_salts") || M.reagents.has_reagent("stimulative_agent") || M.reagents.has_reagent("stimulants"))
+				M.adjustToxLoss(10)
+			else //apply debilitating effects
+				if(prob(75))
+					M.AdjustConfused(5)
+				else
+					M.AdjustWeakened(5)
+		if(44)
+			to_chat(M, "<span class='notice'>Your body goes rigid, you cannot move at all!</span>")
```

Comment:
See above. Also how much does the user get spammed with this message? (Never played with chems, so unsure if this spams or not)

Reply:
This specific message will only occur once (on cycle 44) per application of this chem, assuming the dosage is enough to last 44 cycles.